### PR TITLE
Feature/pull all rows

### PR DESCRIPTION
**Pull request by CloudNiner**

### CODE REVIEW

File: app/scripts/views/download/download-controller.js [hunk 1]
```diff
@@ -57,9 +57,18 @@
...
 
+            var quoteReplaceRe = /"/g;
+
             var csvString = headers.join(',') + '\n';
             angular.forEach(data, function (row) {
-                csvString += _.values(_.omit(row, '$$hashKey')).join(',') + '\n';
+                csvString += _(row).omit('$$hashkey').values().map(function (value) {
+                    if (_.isString(value)) {
+                        // Need to quote strings, and convert double quotes to single quotes
+                        return '"' + value.replace(quoteReplaceRe, '\'') + '"';
```
Comment (sharph) [inline app/scripts/views/download/download-controller.js]: This looks like it will replace double quotes with single quotes.

Comment (CloudNiner) [inline app/scripts/views/download/download-controller.js]: Correct. Double quotes are used by CSV to determine where each column value starts and ends, so if double quotes are left inside the string value the output CSV will be broken. Safest thing to do seemed to be to just swap all double quotes in the string to single quotes.

Comment (sharph) [inline app/scripts/views/download/download-controller.js]: According to http://www.csvreader.com/csv_format.php:

> In an Excel escaped CSV file, in fields containing a double quote, the double quote must be escaped by replacing the single double quote with two double quotes.

But single quotes are fine.

**Comment by sharph** [comment]
+1

File: app/scripts/views/download/download-controller.js (hunk 1 above)
Comment (CloudNiner) [inline app/scripts/views/download/download-controller.js]: I'll swap it to this, since that's the official spec.

**Pull request merged**